scripts/api.js

```diff
@@ -1,17 +1,30 @@
-/*global $ codeUrl */
-/* exported makeApiAjaxSettings */
+/*global codeUrl */
+/* exported ajax */
 
-function makeApiAjaxSettings(apiUrl, queryParams) {
-    let queryParamString = "";
-    if(queryParams) {
-        queryParamString = "&" + $.param(queryParams);
-    }
-    let settings = {
-        "url": codeUrl + "/api/index.php?url=" + apiUrl + queryParamString,
-        "dataType": "json",
-        "headers": {
-            "X-API-KEY": "SESSION"
-        }
+function ajax(method, apiUrl, queryParams = {}, data = {}) {
+    let url = new URL(`${codeUrl}/api/index.php`);
+    queryParams.url = apiUrl;
+    url.search = new URLSearchParams(queryParams);
+    let options = {
+        headers: {"X-API-KEY": "SESSION"},
+        method: method,
     };
-    return settings;
+    if(method !== "GET") {
+        // POST or PUT
+        options.headers['Content-Type'] = 'application/json';
+        options.body = JSON.stringify(data);
+    }
+    return new Promise(function(resolve, reject) {
+        fetch(url, options)
+            .then(function(response) {
+                if(response.ok) {
+                    resolve(response.json());
+                } else {
+                    response.json()
+                        .then(function(data) {
```

Comment:
do we want to catch in case response wasn't json?

Reply:
We probably also need to handle the case where `error` isn't a key in the JSON too. All valid API requests should return one on a non-200 response code but it would be bad to fail on a failure.

Reply:
Should we also look at the response status code? Since we are returning several different codes are there circumstances where we want to look at them and take different actions? WE could have a lower level function that returns the response and wrap it in a higher level function.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
Since 2018 fetch changed the default to include same-origin:
https://github.com/whatwg/fetch/pull/585

Reply:
Wouldn't hurt to put it in anyway.